src/lib.rs

```diff
@@ -1204,6 +1205,56 @@ pub fn is_block_device(device_path: impl AsRef<Path>) -> BlockResult<bool> {
     )))
 }
 
+/// Get property value by key `tag` for device with devpath `device_path` (like "/dev/sda") if present
+#[cfg(target_os = "linux")]
+pub fn get_block_dev_property(
+    device_path: impl AsRef<Path>,
+    tag: &str,
+) -> BlockResult<Option<String>> {
+    let error_message = format!(
```

Comment:
Why incur the mem allocate and overhead for this message before you even have an error?  You could do this `format!` in the closure that creates the `BlockUtilsError`